File: languages/lua/worker.lua
local KEY=0x7d
local function transform(h)local o={} for i=1,#h,2 do local v=tonumber(h:sub(i,i+1),16);o[#o+1]=string.format("%02x",v ~ KEY) end return table.concat(o) end
for line in io.lines() do line=line:gsub("\r$",""); if line=="PING" then print("PONG");io.flush() elseif line=="QUIT" then break elseif line:match("^[ED] [0-9A-Fa-f]*$") then print(transform(line:sub(3)));io.flush() else print("ERR");io.flush() end end

File: languages/elixir/worker.exs
key=0xD5
transform=fn h -> h |> String.graphemes() |> Enum.chunk_every(2) |> Enum.map(fn [a,b] -> {v,""}=Integer.parse(a<>b,16); Integer.to_string(Bitwise.bxor(v,key),16)|>String.pad_leading(2,"0") end) |> Enum.join() end
loop = fn loop -> case IO.read(:stdio,:line) do :eof -> :ok; line -> line=String.trim_trailing(line)|>String.trim_trailing("\r"); cond do line=="PING" -> IO.puts("PONG");loop.(loop); line=="QUIT" -> :ok; String.match?(line,~r/^[ED] [0-9A-Fa-f]*$/) -> h=if String.length(line)>2, do: String.slice(line,2,String.length(line)-2), else: ""; IO.puts(transform.(h));loop.(loop); true -> IO.puts("ERR");loop.(loop) end end end
loop.(loop)
